File: 01.js
// TWO SUM

function twoSum(array, target) { // using a named function
    let indexes = []; // creating an empty array for final result
    for (let i = 0; i < array.length - 1; i++) { // using double loops
        for (let j = i + 1; j < array.length; j++) {
            if (array[i] + array[j] == target) {
                indexes.push(i);
                indexes.push(j); // pushing the indexes of the solutions
            }
        }
    }
    return indexes; // returning the result
}

const result = twoSum([2, 4, 10, 11], 15); // an example
console.log(result); // logging the result

File: 01.py
# TWO SUM

def twoSum(array, target):
    indexes = []
    for i in range(0, len(array), 1):
        for j in range(i + 1, len(array), 1):
            if (array[i] + array[j] == target):
                indexes.append(i)
                indexes.append(j)
    return indexes


result = twoSum([2, 4, 10, 11], 15)
print(result)
